plugin/trino-hudi/src/main/java/io/trino/plugin/hudi/HudiMetadata.java

```diff
@@ -0,0 +1,336 @@
+    {
+        return new ConnectorTableProperties();
+    }
+
+    @Override
+    public Map<String, ColumnHandle> getColumnHandles(ConnectorSession session, ConnectorTableHandle tableHandle)
+    {
+        HudiTableHandle hudiTableHandle = (HudiTableHandle) tableHandle;
+        Table table = metastore.getTable(hudiTableHandle.getSchemaName(), hudiTableHandle.getTableName())
+                .orElseThrow(() -> new TableNotFoundException(schemaTableName(hudiTableHandle.getSchemaName(), hudiTableHandle.getTableName())));
+        return hiveColumnHandles(table, typeManager, NANOSECONDS).stream()
+                .collect(toImmutableMap(HiveColumnHandle::getName, identity()));
+    }
+
+    @Override
+    public ColumnMetadata getColumnMetadata(ConnectorSession session, ConnectorTableHandle tableHandle, ColumnHandle columnHandle)
+    {
+        return ((HiveColumnHandle) columnHandle).getColumnMetadata();
+    }
+
+    @Override
+    public Optional<Object> getInfo(ConnectorTableHandle table)
+    {
+        return ((HudiTableHandle) table).getPartitions()
+                .map(partitions -> new HudiInputInfo(
+                        partitions.stream()
+                                .map(HivePartition::getPartitionId)
+                                .collect(toImmutableList())));
+    }
+
+    @Override
+    public List<SchemaTableName> listTables(ConnectorSession session, Optional<String> optionalSchemaName)
+    {
+        ImmutableList.Builder<SchemaTableName> tableNames = ImmutableList.builder();
+        for (String schemaName : listSchemas(session, optionalSchemaName)) {
+            for (String tableName : metastore.getAllTables(schemaName)) {
+                tableNames.add(schemaTableName(schemaName, tableName));
+            }
+        }
+
+        tableNames.addAll(listMaterializedViews(session, optionalSchemaName));
+        return tableNames.build();
+    }
+
+    @Override
+    public Map<SchemaTableName, List<ColumnMetadata>> listTableColumns(ConnectorSession session, SchemaTablePrefix prefix)
+    {
+        List<SchemaTableName> tables = prefix.getTable()
+                .map(ignored -> singletonList(prefix.toSchemaTableName()))
+                .orElseGet(() -> listTables(session, prefix.getSchema()));
+
+        ImmutableMap.Builder<SchemaTableName, List<ColumnMetadata>> columns = ImmutableMap.builder();
+        for (SchemaTableName table : tables) {
+            try {
+                columns.put(table, getTableMetadata(table).getColumns());
+            }
+            catch (TableNotFoundException e) {
+                // table disappeared during listing operation
+            }
+        }
+        return columns.buildOrThrow();
+    }
+
+    HiveMetastore getMetastore()
+    {
+        return metastore;
+    }
+
+    void rollback()
+    {
+        // TODO: cleanup open transaction when write will be supported
+    }
```

Comment:
I would suggest dropping it and a code which calls to it, and only add it back when writes are actually supported.